volume/local/local.go

```diff
@@ -396,6 +400,33 @@ func (v *localVolume) LiveRestoreVolume(ctx context.Context, _ string) error {
 	return nil
 }
 
+// restoreIfMounted restores the mounted status if the _data directory is already mounted.
+func (v *localVolume) restoreIfMounted() {
+	if v.needsMount() {
+		// Check if the _data directory is already mounted.
+		// Always false on Windows.
+		mounted, err := mountinfo.Mounted(v.path)
+		if err != nil {
+			log.G(context.TODO()).WithFields(log.Fields{
+				"volume": v.name,
+				"path":   v.path,
+				"error":  err,
+			}).Debug("failed to determine if volume is already mounted, assuming no")
+			mounted = false
```

Comment:
This seems redundant, as `mountinfo.Mounted()` would already return `false` on error.

Given that we already have `_unix` and `_windows` implementations; perhaps consider moving this function to the `_unix.go` file, and implement a stub on Windows 🤔

Reply:
^^ I meant `v.needsMount()` (edited my comment)

Reply:
(ignore my previous (deleted) comment, was looking at the wrong spot)

Reply:
The `needsMount` check is still needed here, it's not meant as an indirect check for "is it Windows or not".
It's there to make sure that we only restore the mount status for volume mounts which are actually managed by daemon.
Volumes which are not `v.needsMount()` are not mounted/unmounted by the driver at all, so there's no point in restoring their mounted state.

So it's platform dependent only indirectly.

Reply:
Split the implementation.